src/js/pages/admin.js

```diff
@@ -55,12 +58,31 @@ import { simpleGet } from "../modules/utils.js"
           }
         })
 
-        adminPanelRoot.render(<AdminPanel menus={menus} />)
+        adminPanelRoot.render(<AdminPanel menus={menus} admin={admin} />)
       }
     )
 
+    const auth_api_url = new URL("/auth/api/v1/user-info", baseURL)
+    simpleGet(auth_api_url).then((dataStr) => {
+      try {
+        // check if the response is a valid JSON
+        JSON.parse(dataStr)
+      } catch (e) {
+        console.error("Error parsing auth api JSON response:", e)
+        return
+      }
+      const data = JSON.parse(dataStr)
+      // check if the username is the same as the admin username
+      if (data.username !== admin) {
+        console.error(`User ${data.username} is not the admin user ${admin}.`)
+      }
+      admin.email = data.email
+      admin.name = data.name
+      adminPanelRoot.render(<AdminPanel menus={menus} admin={admin} />)
+    })
+
```

Comment:
I alway try to approach objects like inmutables as most as I can in React, probably for this use case there is no problem, but I think a refactor would be worth. I suggest to move this logic of querying the auth endpoint to the `AdminPanel` react component and have a `useEffect` callback to query the endpoint and also a state variable that stores the name of the user. **I strongly recommend to do this refactor.**

Reply:
That's a fair point. I could move the redis get into the panel too. It would make for cleaner code